src/OmnichannelChatSDK.ts

```diff
@@ -1599,6 +1634,16 @@ class OmnichannelChatSDK {
         }
 
         if (this.liveChatVersion === LiveChatVersion.V2) {
+
+            if (this.isAMSClientAllowed === false) {
+                exceptionThrowers.throwFeatureDisabled(this.scenarioMarker, TelemetryEvent.UploadFileAttachment, "Enable support for attachment upload and receive in the widget configuration.");
+            }
+            const amsClient = await this.getAMSClient();
+
+            if (amsClient === null || amsClient === undefined) {
+                throwAMSLoadFailure(this.scenarioMarker, TelemetryEvent.UploadFileAttachment, "AMS client is null, no action can be performed");
```

Comment:
I’d use “Attachment client” to not expose the name of the internal component publicly

Reply:
I''d like to understand what is the secret we are protecting?? AMS client repo is open source, AMS concepts and exposure is all around the code, which is an open source as well,